### PR TITLE
Add cluster and bucket replication metrics in metrics-v3

### PR DESCRIPTION
**Pull request by anjalshireesh**
## Community Contribution License

All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description

endpoint: /minio/metrics/v3/replication
metrics:
- average_active_workers
- average_queued_bytes
- average_queued_count
- average_data_transfer_rate
- current_active_workers
- current_data_transfer_rate
- last_minute_queued_bytes
- last_minute_queued_count
- max_active_workers
- max_queued_bytes
- max_queued_count
- max_data_transfer_rate

endpoint: /minio/metrics/v3/bucket/replication metrics:
- last_hour_failed_bytes
- last_hour_failed_count
- last_minute_failed_bytes
- last_minute_failed_count
- latency_ms
- proxied_delete_tagging_requests_total
- proxied_get_requests_failures
- proxied_get_requests_total
- proxied_get_tagging_requests_failures
- proxied_get_tagging_requests_total
- proxied_head_requests_failures
- proxied_head_requests_total
- proxied_put_tagging_requests_failures
- proxied_put_tagging_requests_total
- sent_bytes
- sent_count
- total_failed_bytes
- total_failed_count
- proxied_delete_tagging_requests_failures

## Motivation and Context

https://github.com/miniohq/engineering/issues/1510

## How to test this PR?

```
curl "http(s)://<host:port>/minio/metrics/v3/bucket/replication/<bucket>"
curl "http(s)://<host:port>/minio/metrics/v3/replication"
```

Should return the new replication related metrics

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [x] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

**Comment by shtripat** [comment]
Are we still using query string based endpoints? description of PR may need changes.

File: cmd/metrics-v3-api-bucket-replication.go [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+		"Number of failures in PUT tagging requests proxied to replication target",
+		bucketL)
+	bucketReplProxiedPutTaggingRequestsTotalMD = NewCounterMD(bucketReplProxiedPutTaggingRequestsTotal,
+		"Number of PUT tagging requests proxied to replication target",
+		bucketL)
+	bucketReplSentBytesMD = NewCounterMD(bucketReplSentBytes,
+		"Total number of bytes replicated to the target",
+		bucketL)
+	bucketReplSentCountMD = NewCounterMD(bucketReplSentCount,
+		"Total number of bytes replicated to this bucket from another source bucket",
```
Comment (shtripat) [inline cmd/metrics-v3-api-bucket-replication.go]: The two description somehow look confusing to me.
`sent_bytes` says bytes sent to the target, whereas `sent_count` says bytes replicated to this bucket from another source.
cc @poornas

Comment (poornas) [reply cmd/metrics-v3-api-bucket-replication.go]: sent_count is total number of object versions from this bucket to target

Comment (anjalshireesh) [reply cmd/metrics-v3-api-bucket-replication.go]: Thanks, fixed.

**Comment by anjalshireesh** [comment]
> Are we still using query string based endpoints? description of PR may need changes.

@shtripat updated the description with bucket-as-path-component based endpoint. PTAL.

**Review by shtripat — APPROVED** [review]
lgtm

**Review by shtripat — APPROVED** [review]

**Review by poornas — APPROVED** [review]

**Review by poornas — APPROVED** [review]

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**